includes/Core/Assets/Assets.php

```diff
@@ -204,74 +196,43 @@ public function enqueue_asset( $handle ) {
 	 * Enqueues Google fonts.
 	 *
 	 * @since 1.0.0
+	 * @deprecated  n.e.x.t This method is no longer used as fonts are loaded as a normal style dependency now.
 	 */
 	public function enqueue_fonts() {
-		if ( $this->fonts_enqueued ) {
-			return;
+		_deprecated_function( __METHOD__, 'n.e.x.t' );
+
+		$assets = $this->get_assets();
+
+		if ( ! empty( $assets['googlesitekit-fonts'] ) && $assets['googlesitekit-fonts'] instanceof Asset ) {
+			$assets['googlesitekit-fonts']->enqueue();
 		}
+	}
 
+	/**
+	 * Get Google fonts src for CSS.
+	 *
+	 * @since n.e.x.t
+	 * @return false|string String URL src, or `false` if no font families to load.
+	 */
+	protected function get_fonts_src() {
 		$font_families = array(
 			'Google+Sans:300,300i,400,400i,500,500i,700,700i',
 			'Roboto:300,300i,400,400i,500,500i,700,700i',
 		);
 
 		$filtered_font_families = apply_filters( 'googlesitekit_font_families', $font_families );
 
-		if ( ! is_array( $filtered_font_families ) || empty( $filtered_font_families ) ) {
-			return;
+		if ( empty( $filtered_font_families ) ) {
+			return false;
```

Comment:
Nit-pick, but maybe let's make this an empty string? While this works in PHP, I'm always in favor of using one return type where possible, and both `false` and `''` are false-y anyway.